src/stravalib/client.py

```diff
@@ -638,12 +639,85 @@ def get_friend_activities(self, limit: int | None = None) -> NoReturn:
+            activity.
+        activity_type : str, default=None
+            The activity type (case-insensitive).
+            Deprecated. Prefer to use sport_type. In a request where both type
+            and sport_type are present, this field will be ignored.
+            See https://developers.strava.com/docs/reference/#api-models-UpdatableActivity.
+            Possible values: ride, run, swim, workout, hike, walk, nordicski,
+            alpineski, backcountryski, iceskate, inlineskate, kitesurf,
+            rollerski, windsurf, workout, snowboard, snowshoe
+        sport_type : str, default=None
+            Possible values (case-sensitive): AlpineSki, BackcountrySki,
+            Badminton, Canoeing, Crossfit, EBikeRide, Elliptical,
+            EMountainBikeRide, Golf, GravelRide, Handcycle,
+            HighIntensityIntervalTraining, Hike, IceSkate, InlineSkate,
+            Kayaking, Kitesurf, MountainBikeRide, NordicSki, Pickleball,
+            Pilates, Racquetball, Ride, RockClimbing, RollerSki, Rowing, Run,
+            Sail, Skateboard, Snowboard, Snowshoe, Soccer, Squash,
+            StairStepper, StandUpPaddling, Surfing, Swim, TableTennis, Tennis,
+            TrailRun, Velomobile, VirtualRide, VirtualRow, VirtualRun, Walk,
+            WeightTraining, Wheelchair, Windsurf, Workout, Yoga
+
+        Returns
+        -------
+        dict
+            Dictionary containing overlapping parameter values between the two
+            methods
+        Raises
+        ------
+        ValueError
+            If the activity_type or sport_type is invalid.
+        """
+
+        # Handle activity type throwing warning given sport type is preferred
+        # This also ONLY adds activity type if the user wants to use it
+        # thus we won't need the "pop" from the dict anymore
+        if activity_type is not None:
+            if not activity_type.lower() in [
+                t.lower() for t in model.Activity.TYPES
+            ]:
+                raise ValueError(
+                    f"Invalid activity type: {activity_type}. Possible values: {model.Activity.TYPES!r}"
+                )
+            params["type"] = activity_type.lower()
+            warn_param_deprecation(
+                "activity_type",
+                "sport_type",
+                "https://developers.strava.com/docs/reference/#api-models-UpdatableActivity",
+            )
+
+        if sport_type is not None:
+            if not sport_type in model.Activity.SPORT_TYPES:
+                raise ValueError(
+                    f"Invalid activity type: {sport_type}. Possible values: {model.Activity.SPORT_TYPES!r}"
+                )
+            params["sport_type"] = sport_type
+
+        return params
+
+    # TODO: according to the strava api docs we can add an optional trainer
+    # and commute param here with boolean 0/1 values
     def create_activity(
         self,
         name: str,
-        activity_type: ActivityType,
```

Comment:
ok. i didn't consider that but you are absolutely right. Do you agree however that it is what we should do if strava is moving away from activity type / type and over to sport_type? 

i'll make a note in the changelog that this is a breaking change now

Reply:
i added this to the changelog for now:

> - Fix: Add sport type to create_activity and create type validator method. NOTE: this fix contains a breaking change in `Client.create_activity()` activity_type is now an optional keyword argument rather than a required positional argument (@lwasser, #279)